type.go

```diff
@@ -309,10 +310,11 @@ func (t *intType) ConvertedType() *deprecated.ConvertedType {
 //
 // https://github.com/apache/parquet-format/blob/master/LogicalTypes.md#decimal
 func Decimal(scale, precision int, typ Type) Node {
-	switch typ {
-	case Int32Type, Int64Type:
+	length := int(math.Ceil((math.Log10(2) + float64(precision)) / math.Log10(256)))
+	switch typ.Kind() {
+	case Int32Type.Kind(), Int64Type.Kind(), FixedLenByteArrayType(length).Kind():
```

Comment:
This might be a simpler form of the same behavior?
```suggestion
	case Int32, Int64, FixedLenByteArray:
```